qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -149,12 +149,15 @@ def _run_analysis(self, experiment_data: ExperimentData):
             # Since copy for replace result is handled at the parent level
             # we always run with replace result on component analysis
             self._analyses[i].run(sub_expdata, replace_results=True)
-
-        # Analysis is running in parallel so we add loop to wait
-        # for all component analysis to finish before returning
-        # the parent experiment analysis results
-        for sub_expdata in component_expdata:
+            # Block for results to avoid issues nested CompositeAnalysis runs.
```

Comment:
I think this change is reasonable, but should be separate PR (based on the rule of one logical change per PR). Probably we can add composite analysis option to switch this behavior.

Reply:
Not sure how this is added, I didn't add it?